pkg/resources/pciNetDevice.go

```diff
@@ -30,11 +30,25 @@ func NewPciNetDevice(pciDevice *ghw.PCIDevice, rFactory types.ResourceFactory) (
 
 	// 			1. get PF details, add PF info in its pciNetDevice struct
 	// 			2. Get driver info
+	var ifName string
 	pciAddr := pciDevice.Address
 	driverName, err := utils.GetDriverName(pciAddr)
 	if err != nil {
 		return nil, err
 	}
+	netDevs, err := utils.GetNetNames(pciAddr)
+	if err != nil {
+		return nil, err
```

Comment:
We need to consider some VF may not have a kernel netdev name(with uio, vfio drivers attached). In those cases we still want to add the devices into the device list. Maybe ignore the error here and only add ifname if it is available.

Reply:
won't this make errors or bugs with other devices types?

Reply:
> won't this make errors or bugs with other devices types?

This will break existing vfio/uio device support. What type of bugs/errors are we concerned about?

Reply:
fixed